### PR TITLE
CO-262 clear stale CodeRabbit rereview latch

### PR DESCRIPTION
**Pull request by Kbediako**
## What

- Add CO-262 docs-first packet and registry mirrors for the ready-review CodeRabbit rereview latch bug.
- Teach `pr ready-review` to distinguish raw and effective bot rereview pending state.
- Clear stale `coderabbitai` rereview pending only when recognized current-head CodeRabbit `statusCheckRollup` evidence is explicit `SUCCESS`, newer than the latest rereview request, review threads are resolved, and current-head bot feedback truth is clean.
- Keep valid pending, failed, skipped, stale, unrelated, unresolved-thread, unacknowledged-feedback, and merge-state blockers explicit in diagnostics and status output.

## Why

CO-223 / PR #556 and CO-260 / PR #555 exposed a stale local CodeRabbit rereview latch where GitHub current-head truth was already clean. This keeps CodeRabbit gating intact while letting current-head `CodeRabbit SUCCESS` plus clean thread/feedback truth clear only the stale latch.

## Validation

- `npx vitest run tests/pr-watch-merge.spec.ts`
- `npx vitest run --config vitest.config.core.ts tests/pr-watch-merge.spec.ts`
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint` (passes with pre-existing `DelegationMcpHealth.test.ts` warnings)
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`
- `npm run pack:smoke`
- Manifest-backed review: `review_outcome: bounded-success`
- Post-merge elegance review: no additional simplification patch needed

Linear: CO-262

## Summary by CodeRabbit

* **New Features**
  * Ready-review latch now clears stale CodeRabbit rereview blocks when the current PR head shows CodeRabbit SUCCESS, zero unresolved review threads, and clean merge state.
  * Added richer diagnostics for rereview status, latch-clear events and gating reasons.

* **Tests**
  * Expanded test coverage for ready-review gating, CodeRabbit success/pending cases, merge-state interactions and edge conditions.

### CODE REVIEW

File: tasks/index.json [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+          "source_anchor": "ctx:sha256:da19ec03e5c89cbe79e6e15120de3e9039ee81e75235ba564c59bb026184f5a2#chunk:c000001",
+          "summary": "Docs-first packet preserves CO-262 as the ready-review stale CodeRabbit rereview latch fix: current-head statusCheckRollup CodeRabbit SUCCESS plus resolved review threads and clean feedback truth should clear stale local CodeRabbit rereview pending, while pending or failed CodeRabbit, unresolved review threads, unacknowledged bot feedback, and merge-state blockers remain explicit readiness gates."
+        }
+      },
+      "paths": {
+        "spec": "tasks/specs/linear-12bea7bd-4716-4c9a-a49c-74d29151bedf.md",
+        "task": "tasks/tasks-linear-12bea7bd-4716-4c9a-a49c-74d29151bedf.md",
+        "agent_task": ".agent/task/linear-12bea7bd-4716-4c9a-a49c-74d29151bedf.md",
+        "docs": "docs/TECH_SPEC-linear-12bea7bd-4716-4c9a-a49c-74d29151bedf.md"
+      }
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/index.json:29]: @coderabbitai Fixed in 356ffc2d6: changed this entry to the canonical paths.docs.TECH_SPEC shape. Please re-review and resolve.

File: tasks/specs/linear-12bea7bd-4716-4c9a-a49c-74d29151bedf.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+- `unresolved review threads` is zero.
+- `mergeStateStatus CLEAN` is present.
+- No current-head actionable review blocker remains.
+
+The parent implementation should keep blocking when:
+
+- CodeRabbit is missing, pending, failed, or stale.
+- `unresolved review threads` is nonzero.
+- merge state is not clean.
+- actionable current-head review feedback remains.
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/specs/linear-12bea7bd-4716-4c9a-a49c-74d29151bedf.md:67]: @coderabbitai Fixed in 356ffc2d6: decoupled merge state from the CodeRabbit latch-clear condition and documented mergeStateStatus as a separate readiness gate. Please re-review and resolve.
